app/scripts/clean_data.py

```diff
@@ -0,0 +1,153 @@
+
+        # Store supported locales for this repository
+        locales_file = os.path.join(sources_path, repository_id + '.txt')
+        supported_locales = open(locales_file, 'r').read().splitlines()
+
+        supported_repositories[repository_id] = {
+            'folder_name': folder_name,
+            'locales': supported_locales
+        }
+
+        # Store a list of acceptable cache file names
+        for locale in supported_locales:
+            known_cache_files.append(
+                '{0}/{1}/cache_{1}_{2}.php'.format(storage_path, locale, repository_id))
+
+    # List all .txt files in /sources
+    print '--\nAnalyzing sources in config/sources'
+    excluded_files = ['gaia_versions']
+
+    need_cleanup = False
+    for txtfile in glob.glob(os.path.join(sources_path, '*.txt')):
+        filename = os.path.splitext(os.path.basename(txtfile))[0]
+        if not filename in supported_repositories.keys() + excluded_files:
+            print '{0}.txt is not a supported repository.'.format(filename)
+            need_cleanup = True
+            if args.delete:
+                print "Removing file:", txtfile
+                os.remove(txtfile)
+    if not need_cleanup:
+        print "Nothing to remove."
+
+    # Check all repositories for extra folders
+    print '--\nAnalyzing folders in supported repositories'
+    exclusions = {
+        'firefox_ios': ['.git', 'templates'],
+        'mozilla_org': ['.git', 'en-US', 'fur', 'hi']
+    }
+    hg_path = config_parser.get('config', 'local_hg')
+    git_path = config_parser.get('config', 'local_git')
+
+    need_cleanup = False
+    for repository_id, repository in supported_repositories.iteritems():
+        # Check if the folder exists as a Mercurial repository. If it doesn't
+        # assume it's a Git repository.
+        print '--\nAnalyze', repository_id
+        if os.path.isdir(os.path.join(hg_path, repository['folder_name'])):
+            folder_path = os.path.join(hg_path, repository['folder_name'])
+        else:
+            folder_path = os.path.join(git_path, repository['folder_name'])
+
+        available_folders = os.walk(folder_path).next()[1]
+        available_folders.sort()
+        for folder in available_folders:
+            if folder in exclusions.get(repository_id, []):
+                continue
+            if not folder in repository['locales']:
+                # This folder is inside the repository but doesn't match
+                # any supported locale.
+                print '{0} is not a supported locale'.format(folder)
+                need_cleanup = True
+                if args.delete:
+                    full_path = os.path.join(folder_path, folder)
+                    print "Removing folder:", full_path
+                    shutil.rmtree(full_path)
+    if not need_cleanup:
+        print "Nothing to remove."
+
+    # Check cache files
+    print '--\nAnalyze cache files in TMX'
+    excluded_folders = [
```

Comment:
I think it would be valuable to put a comment here explaining why we exclude fur and hi